File: 프로그래머스/2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dict:[String:Int] = [:]
    for c in clothes {
        if !dict.keys.contains(c[1]) { dict[c[1]] = 1 }
        else { dict[c[1]]! += 1 }
    }

    var result = 1
    for d in dict.keys { result *= dict[d]!+1 }
    return result-1
}

File: 프로그래머스/2/42578. 의상/의상.py
from itertools import combinations

def solution(clothes):
    dict = {}
    result = 1

    for c in clothes:
        if not c[1] in dict.keys(): dict[c[1]] = 0
        dict[c[1]] += 1
        
    for k in dict.keys(): result *= dict[k] + 1
    return result-1